src/Microsoft.Diagnostics.Runtime/datatarget.cs

```diff
@@ -2045,16 +2058,40 @@ public unsafe bool GetThreadContext(uint threadID, uint contextFlags, uint conte
     internal unsafe class LiveDataReader : IDataReader
     {
         #region Variables
+        private int _originalPid;
+        private IntPtr _snapshotHandle;
+        private IntPtr _vaCloneHandle;
         private IntPtr _process;
         private int _pid;
         #endregion
 
         private const int PROCESS_VM_READ = 0x10;
         private const int PROCESS_QUERY_INFORMATION = 0x0400;
-        public LiveDataReader(int pid)
+        public LiveDataReader(int pid, bool createSnapshot = false)
         {
-            _pid = pid;
-            _process = OpenProcess(PROCESS_VM_READ | PROCESS_QUERY_INFORMATION, false, pid);
+            if (createSnapshot)
+            {
+                _originalPid = pid;
+                var process = Process.GetProcessById(pid);
```

Comment:
Can you change these 'var' statements to use the actual types?  (Throughout this change.)  It's not completely consistent throughout the code, but the only time var should be used is generally in statements where it reduces a ton of redundancy in a new statement.  Having 'Process process', 'int hr' makes it clear what we are working with.